cpu/arm7tdmi_gba/include/periph_cpu.h

```diff
@@ -0,0 +1,120 @@
+/**
+ * @ingroup         cpu_arm7tdmi_gba
+ * @{
+ *
+ * @file
+ * @brief           CPU specific definitions for internal peripheral handling
+ *
+ * @author
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#define PM_NUM_MODES 0
+
+/**
+ * @brief   Type for 32-bit registers
+ */
+#define REG32(ADDR) (*(uint32_t volatile (*))(ADDR))
+
+/**
+ * @brief   Type for 16-bit registers
+ */
+#define REG16(ADDR) (*(uint16_t volatile (*))(ADDR))
+
+/**
+ * @brief   Type for 16-bit memory addresses
+ */
+#define MEM16(ADDR) ((uint16_t(*))(ADDR))
+
+#define GBA_IRQ_ISR 0x03FFFFFC
+#define GBA_IRQ_BIOS_ACK REG16(0x03FFFFF8)
+
+#define GBA_DISPCNT REG16(0x4000000)
+
+#define GBA_DISPCNT_BGMODE_0 0x0000
+#define GBA_DISPCNT_BGMODE_1 0x0001
+#define GBA_DISPCNT_BGMODE_2 0x0002
+/* Bitmap, 240x160, 15-bit RGB */
+#define GBA_DISPCNT_BGMODE_3 0x0003
+
+
+#define GBA_DISPCNT_SDBG_0 0x0100
+#define GBA_DISPCNT_SDBG_1 0x0200
+/* Display BG2 */
+#define GBA_DISPCNT_SDBG_2 0x0400
+
+#define GBA_DISPSTAT REG16(0x4000004)
+
+#define GBA_BG0CNT REG16(0x4000008)
+#define GBA_BG1CNT REG16(0x400000a)
+#define GBA_BG2CNT REG16(0x400000c)
+#define GBA_BG3CNT REG16(0x400000e)
+
+#define GBA_KEYPAD REG16(0x4000130)
+#define GBA_KEYPAD_IRQCNTL REG16(0x4000132)
+#define GBA_KEY_A 0x0001
+#define GBA_KEY_B 0x0002
+#define GBA_KEY_SELECT 0x0004
+#define GBA_KEY_START 0x0008
+#define GBA_KEY_RIGHT 0x0010
+#define GBA_KEY_LEFT 0x0020
+#define GBA_KEY_UP 0x0040
```

Comment:
HW specific register defines should go to their own file, `periph_cpu.h` is for common (RIOT) interface config